src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletRPCControllerTest.cs

```diff
@@ -0,0 +1,696 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Flurl;
+using Flurl.Http;
+using NBitcoin;
+using Stratis.Bitcoin.Controllers.Models;
+using Stratis.Bitcoin.Features.BlockStore.Models;
+using Stratis.Bitcoin.Features.RPC;
+using Stratis.Bitcoin.Features.Wallet.Models;
+using Stratis.Bitcoin.IntegrationTests.Common;
+using Stratis.Bitcoin.IntegrationTests.Common.EnvironmentMockUpHelpers;
+using Stratis.Bitcoin.IntegrationTests.Common.ReadyData;
+using Stratis.Bitcoin.Networks;
+using Xunit;
+
+namespace Stratis.Bitcoin.IntegrationTests.Wallet
+{
+    public class WalletRPCControllerTest
```

Comment:
Very nice set of tests... wondering whether we should be moving some of the tests from Stratis.Bitcoin.IntegrationTests.RPC related to the WalletRPCController to this class? I could do that in a separate PR later if you think it's a good idea. Does WithBlockChainReadyData() avoid the need to use "generate" to make some transactions?

Reply:
Yeah these tests should cover everything really.
Well, this is RPC and wallet so it's hard to know whether they should go with RPC or wallet.
I'm gonna go through all these methods now to triple check they're ok and if I write some tests I'll drop them in there.

With WithBlockChainReadyData, what we did is that we created a blockchain with 10/100/150 blocks already mined, so that we don't have to mine every time we need some data. We've gained something like 45 minutes on the build already.